File: src/main/kotlin/com/chopeks/formatter/settings/SquirrelCodeStyleSettingsProvider.java
package com.chopeks.formatter.settings;

import com.chopeks.SquirrelBundle;
import com.intellij.openapi.options.Configurable;
import com.intellij.psi.codeStyle.CodeStyleSettings;
import com.intellij.psi.codeStyle.CodeStyleSettingsProvider;
import com.intellij.psi.codeStyle.CustomCodeStyleSettings;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class SquirrelCodeStyleSettingsProvider extends CodeStyleSettingsProvider {
  @Override
  public String getConfigurableDisplayName() {
    return SquirrelBundle.message("squirrel.title");
  }

  @NotNull
  @Override
  public Configurable createSettingsPage(CodeStyleSettings settings, CodeStyleSettings originalSettings) {
    return new SquirrelCodeStyleConfigurable(settings, originalSettings);
  }

  @Nullable
  @Override
  public CustomCodeStyleSettings createCustomSettings(CodeStyleSettings settings) {
    return new SquirrelCodeStyleSettings(settings);
  }
}

File: src/main/kotlin/com/chopeks/formatter/settings/SquirrelCodeStyleSettingsProvider.kt
package com.chopeks.formatter.settings

import com.chopeks.SquirrelBundle.message
import com.intellij.psi.codeStyle.CodeStyleSettings
import com.intellij.psi.codeStyle.CodeStyleSettingsProvider

class SquirrelCodeStyleSettingsProvider : CodeStyleSettingsProvider() {
	override fun getConfigurableDisplayName() = message("squirrel.title")
	override fun createCustomSettings(settings: CodeStyleSettings) = SquirrelCodeStyleSettings(settings)
	override fun createSettingsPage(settings: CodeStyleSettings, originalSettings: CodeStyleSettings) =
		SquirrelCodeStyleConfigurable(settings, originalSettings)
}
